packages/pluggableWidgets/language-selector-web/src/components/LanguageSwitcher.tsx

```diff
@@ -0,0 +1,147 @@
+import classNames from "classnames";
+import { createElement, ReactElement, useEffect, useRef, useState, useCallback } from "react";
+import {
+    isBehindElement,
+    isBehindRandomElement,
+    isElementPartiallyOffScreen,
+    isElementVisibleByUser,
+    moveAbsoluteElementOnScreen,
+    unBlockAbsoluteElementBottom,
+    unBlockAbsoluteElementLeft,
+    unBlockAbsoluteElementRight,
+    unBlockAbsoluteElementTop,
+    handleOnClickOutsideElement
+} from "@mendix/piw-utils-internal";
+import { PositionEnum, TriggerEnum } from "../../typings/LanguageSelectorProps";
+import { LanguageItem } from "../LanguageSelector";
+
+export interface LanguageSwitcherProps {
+    preview: boolean;
+    currentLanguage: LanguageItem | undefined;
+    languageList: LanguageItem[];
+    position: PositionEnum;
+    onSelect?: (lang: LanguageItem) => void;
+    trigger: TriggerEnum;
+    className: string;
+}
+export const LanguageSwitcher = (props: LanguageSwitcherProps): ReactElement => {
+    const preview = !!props.preview;
+    const { languageList, trigger } = props;
+    const [visibility, setVisibility] = useState(false);
+    const ref = useRef<HTMLDivElement>(null);
+
+    if (!preview) {
+        handleOnClickOutsideElement(ref, () => setVisibility(false));
+    }
+
+    const menuOptions = (): ReactElement[] => {
```

Comment:
Is there a reason to have this as a function?

Reply:
I believe this is a legacy from Popup :(